File: python/api-calls/media_output_get.py
#!/usr/bin/env python

import os
import shutil
import requests

output_path = os.environ["OUTPUT_MEDIA_LOCAL_PATH"]

url = "https://api.dolby.com/media/output"
headers = {
    "x-api-key": os.environ["DOLBYIO_API_KEY"],
    "Content-Type": "application/json",
    "Accept": "application/json",
}

args = {
    "url": "dlb://out/example-enhanced.mp3",
}

with requests.get(url, params=args, headers=headers, stream=True) as response:
    response.raise_for_status()
    response.raw.decode_content = True
    print("Downloading from {0} into {1}".format(response.url, output_path))
    with open(output_path, "wb") as output_file:
        shutil.copyfileobj(response.raw, output_file)

File: javascript/api-calls/media_output_get.js
#!/usr/bin/env node

const fs = require('fs');
const axios = require('axios').default;

const output_path = process.env.OUTPUT_MEDIA_LOCAL_PATH;

const config = {
    method: 'get',
    url: 'https://api.dolby.com/media/output',
    headers: {
        'x-api-key': process.env.DOLBYIO_API_KEY,
        'Content-Type': 'application/json',
        'Accept': 'application/json'
    },
    responseType: 'stream',
    params: {
        url: 'dlb://out/example-enhanced.mp3'
    }
};

axios(config)
    .then(function (response) {
        response.data.pipe(fs.createWriteStream(output_path));
        response.data.on('error', function (error) {
            console.log(error);
        });
        response.data.on('end', function () {
            console.log('File downloaded!');
        });
    })
    .catch(function (error) {
        console.log(error);
    });
